packages/web-pkg/src/services/loadingService.ts

```diff
@@ -0,0 +1,100 @@
+import * as uuid from 'uuid'
+import { eventBus } from 'web-pkg'
+import { debounce } from 'lodash-es'
+
+export enum LoadingEventTopics {
+  add = 'loading-service.add',
+  remove = 'loading-service.remove',
+  setProgress = 'loading-service.set-progress'
+}
+
+export interface LoadingTaskState {
+  total: number
+  current: number
+}
+
+export interface LoadingTask {
+  id: string
+  active: boolean
+  state?: LoadingTaskState
+}
+
+// time until a loading task is being set active
+const DEFAULT_DEBOUNCE_TIME = 200
+
+export class LoadingService {
+  private tasks: LoadingTask[] = []
+
+  public get isLoading(): boolean {
+    return this.tasks.some((e) => e.active)
+  }
+
+  /**
+   * Get the current progress from 0 to 100.
+   * Returns null if at least one task is indeterminate.
+   */
+  public get currentProgress(): number | null {
+    if (this.tasks.some((e) => !e.state && e.active)) {
+      return null
+    }
+
+    const tasks = this.tasks.filter((e) => !!e.state && e.active)
+    if (!tasks.length) {
+      return null
+    }
+
+    const progress = tasks.reduce((acc, task) => {
+      acc += task.state.current / task.state.total
+      return acc
+    }, 0)
+
+    return Math.round((progress / tasks.length) * 100)
+  }
+
+  public addTask(
+    callback: ({ setProgress }: { setProgress: (args: LoadingTaskState) => void }) => Promise<any>,
+    {
+      debounceTime = DEFAULT_DEBOUNCE_TIME,
+      indeterminate = true
+    }: { debounceTime?: number; indeterminate?: boolean } = {}
+  ): Promise<any> {
```

Comment:
I think, we can improve the types a little:

```typescript
// not a great name, better ideas appreciated
// maybe `LoadingTaskProgress` and then a `.set` method?
export interface LoadingTaskCallbackArguments {
  setProgress: (args: LoadingTaskState) => void
}
// …
  public addTask<T>(
    callback: ({ setProgress }: LoadingTaskCallbackArguments) => Promise<T>,
    {
      debounceTime = DEFAULT_DEBOUNCE_TIME,
      indeterminate = true
    }: { debounceTime?: number; indeterminate?: boolean } = {}
  ): Promise<T> {
  // …
  }
```

That way 
- we can more easily pass on the `LoadingTaskCallbackArguments` to helper classes/functions, which I prefer over just passing around the `setProgress` function, as it makes it easier to add more features later (`.advanceOne()`, `.advance(2)` or something like that e.g.  :)) 
- we (more importantly) don't lose the return type of the callback function